repository/src/lib.rs

```diff
@@ -439,10 +439,67 @@ impl<T: RawRepository> DistributedRepository<T> {
     /// and update the corresponding `a-#` branch to it
     pub async fn approve(
         &mut self,
-        _agenda_commit_hash: &CommitHash,
-        _proof: Vec<(PublicKey, TypedSignature<Agenda>)>,
+        agenda_commit_hash: &CommitHash,
+        proof: Vec<(PublicKey, TypedSignature<Agenda>)>,
     ) -> Result<CommitHash, Error> {
-        unimplemented!()
+        let last_header = self.get_last_finalized_block_header().await?;
+        let last_header_commit = self.raw.locate_branch(FINALIZED_BRANCH_NAME.into()).await?;
+        // Check if the agenda is rebased on top of the `finalized` branch.
+        if self
+            .raw
+            .find_merge_base(last_header_commit, *agenda_commit_hash)
+            .await?
+            != last_header_commit
+        {
+            return Err(anyhow!(
+                "agenda:{} should be rebased on {}",
+                agenda_commit_hash,
```

Comment:
```suggestion
                "branch {} should be rebased on {}",
                agenda_branch_name,
```
keep the error message same with other methods please

Reply:
fixed.